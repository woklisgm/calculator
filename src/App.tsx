import React from 'react';

import styles from './app.css';
import { Calculator } from './components/Calculator';
// import './fonts/Geometria/stylesheet.css';

function App() {

	return (
		<div className={styles.app}>
			<div className={styles.wrapper}>
				<Calculator />
			</div>
		</div>
	);
}

export default App;
