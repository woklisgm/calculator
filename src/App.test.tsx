import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer  from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import App from './App';
import { renderWithRedux } from './tests/helpers/renderWithRedux';

const mockCheckKey = jest.fn();

jest.mock('./hooks/useActions', () => ({
	useActions: () => ({
		checkKey: mockCheckKey
	})
}));

describe('App test', () => {
	test('snapshot', () => {
		const treeApp = renderer.create(renderWithRedux(<App />)).toJSON();

		expect(treeApp).toMatchSnapshot();
	});

	test('check keyboard press', async () => {
		render(renderWithRedux(<App />));
		
		userEvent.keyboard('1');
		expect(mockCheckKey).toBeCalledTimes(1);
	});

	test('calculator in the document', () => {
		render(renderWithRedux(<App />));
		
		const el = screen.getByTestId('calculator');
		expect(el).toBeInTheDocument();
	})

	test('keyboard in the document', () => {
		render(renderWithRedux(<App />));
		
		const el = screen.getByTestId('keyboard');
		expect(el).toBeInTheDocument();
	})

	test('display in the document', () => {
		render(renderWithRedux(<App />));
		
		const el = screen.getByTestId('keyboard');
		expect(el).toBeInTheDocument();
	})
})

// jest.mock('./hooks/useActions', () => {
// 	jest.fn(() => {
// 		return {
// 			__esModule: true,
// 			checkKey: jest.fn(() => true)
// 		}
// 	})
// });