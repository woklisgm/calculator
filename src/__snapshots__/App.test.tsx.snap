// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test snapshot 1`] = `
<div
  className="app"
>
  <div
    className="wrapper"
  >
    <div
      className="calculator"
      data-testid="calculator"
    >
      <div
        className="display"
        data-testid="display"
      >
        <div
          className="expression"
        >
          <span
            className=""
            data-testid="expression"
          >
            <i />
          </span>
        </div>
        <div
          className="result"
          data-testid="result"
        >
          0
        </div>
      </div>
      <div
        className="keyboard"
        data-testid="keyboard"
        onClick={[Function]}
      >
        <div
          className="btn"
          data-char="C"
          data-testid="keyboard-btn"
        >
          <div>
            C
          </div>
        </div>
        <div
          className="btn"
          data-char="√"
          data-testid="keyboard-btn"
        >
          <div>
            √
          </div>
        </div>
        <div
          className="btn"
          data-char="%"
          data-testid="keyboard-btn"
        >
          <div>
            %
          </div>
        </div>
        <div
          className="btn"
          data-char="/"
          data-testid="keyboard-btn"
        >
          <div>
            /
          </div>
        </div>
        <div
          className="btn"
          data-char="7"
          data-testid="keyboard-btn"
        >
          <div>
            7
          </div>
        </div>
        <div
          className="btn"
          data-char="8"
          data-testid="keyboard-btn"
        >
          <div>
            8
          </div>
        </div>
        <div
          className="btn"
          data-char="9"
          data-testid="keyboard-btn"
        >
          <div>
            9
          </div>
        </div>
        <div
          className="btn"
          data-char="×"
          data-testid="keyboard-btn"
        >
          <div>
            ×
          </div>
        </div>
        <div
          className="btn"
          data-char="4"
          data-testid="keyboard-btn"
        >
          <div>
            4
          </div>
        </div>
        <div
          className="btn"
          data-char="5"
          data-testid="keyboard-btn"
        >
          <div>
            5
          </div>
        </div>
        <div
          className="btn"
          data-char="6"
          data-testid="keyboard-btn"
        >
          <div>
            6
          </div>
        </div>
        <div
          className="btn"
          data-char="-"
          data-testid="keyboard-btn"
        >
          <div>
            -
          </div>
        </div>
        <div
          className="btn"
          data-char="1"
          data-testid="keyboard-btn"
        >
          <div>
            1
          </div>
        </div>
        <div
          className="btn"
          data-char="2"
          data-testid="keyboard-btn"
        >
          <div>
            2
          </div>
        </div>
        <div
          className="btn"
          data-char="3"
          data-testid="keyboard-btn"
        >
          <div>
            3
          </div>
        </div>
        <div
          className="btn"
          data-char="+"
          data-testid="keyboard-btn"
        >
          <div>
            +
          </div>
        </div>
        <div
          className="btn"
          data-char="00"
          data-testid="keyboard-btn"
        >
          <div>
            00
          </div>
        </div>
        <div
          className="btn"
          data-char="0"
          data-testid="keyboard-btn"
        >
          <div>
            0
          </div>
        </div>
        <div
          className="btn"
          data-char=","
          data-testid="keyboard-btn"
        >
          <div>
            ,
          </div>
        </div>
        <div
          className="btn"
          data-char="="
          data-testid="keyboard-btn"
        >
          <div>
            =
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
